### ISSUE TITLE
[Bug] Carousel View Current Item, Item Changed, Position Changed Event / Command are updated at wrong time

### ISSUE THREAD

**Issue by muhaym** [xamarin/Xamarin.Forms#10472]
### Description

Carousel View in Snapping Mode Mandatory, and Swipe Mode True User expect, the events Item Changed, Position Changed occurs only after the swipe is completed and the next item is ready and snapped completely.

### Steps to Reproduce

1. Create Carousel View with any Layout that takes full screen for each item
2. Set Snapping Mode Mandatory, Snap Position at End
3. IsSwipeEnabledshould be True 

### Expected Behavior
CurrentItem, PositionChangedCommand, CurrentItemChangedCommand, and corresponding events should only fire after the item is completely snapped

### Actual Behavior
CurrentItem, PositionChangedCommand, CurrentItemChangedCommand, and corresponding events are changed / fired when swiping and view reaching in the middle.

**Comment by rmarinho** [comment]
Hi @muhaym can you tell me why you will expect the events  to fire before the item is snapped?

**Comment by muhaym** [comment]
@rmarinho  I'm not expecting the events or changes before snapping. I'm expecting it after completing the snapping. 

To be precise, the current scenario is, if you have a full page carousel, and IsSwipeEnabled = true, you swipe the element till **the middle** (where you can see both items now), CurrentItem, PositionChangedCommand, CurrentItemChangedCommand, and corresponding events are fired.

But I believe the expected behaviour should be CurrentItem, and other corresponding events should only be changed / fire after the swipe is completed. That is when the new item is completely in view or actual snapping is completed.

**Comment by rmarinho** [comment]
CarouselView is based on always centering the item, and item on the Center is the CurrentItem, so when you want to do animations for example when it comes into view. 

I think what we could do it's maybe an API to aloow to only update when snap happens.

**Comment by muhaym** [comment]
Yeah, try this repro https://github.com/muhaym/ReproVSMCarousel

To explain the use cases, think of App like Tiktok, where Video is swiped continously to consume next item. But the current video is played till it's out of the view completely, and next item is starts to play. With current Carousel View such use-case cannot be achieved. (Or I'm not aware of any work arounds to get similar effect).

https://github.com/muhaym/ReproVSMCarousel This repro I made for another issue, but the same can be used to understand the issue. You can notice the background colour changing when the swipe reaches in the middle, actually I expect the Current Item to be changed only after the swipe is completed.

**Comment by muhaym** [comment]
@davidortinau With the new blog post up https://devblogs.microsoft.com/xamarin/xamarin-forms-5-preview/ I request you to take a look at this issue.

Let's say the attached video to present carousel view, say you had a video in that list like Instagram, you wouldn't want to start playing it till it's completely in view. The current limitation doesn't allow to handle such cases.
